src/events/AdminEvents.tsx

```diff
@@ -540,6 +541,19 @@ export const AdminEvents = () => {
     [t("ipAddress"), authEvent?.authDetails?.ipAddress],
   ];
 
+  function prettyPrintJSON(json: string) {
+    try {
+      return JSON.stringify(JSON.parse(json), null, 2);
+    } catch (error) {
+      return json;
+    }
+  }
+
+  const code = useMemo(
+    () => prettyPrintJSON(representationEvent?.representation!),
```

Comment:
It might be better to make this conditional so there is no need for a non-null assertion (as this might produce a runtime error).
```suggestion
    () => prettyPrintJSON(representationEvent?.representation ? prettyPrintJSON(representationEvent?.representation) : undefined),
```